eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java

```diff
@@ -23,76 +23,64 @@
  */
 package org.eolang.maven.name;
 
-import java.util.Map;
+import org.cactoos.scalar.Unchecked;
 import org.eolang.maven.hash.CommitHash;
-import org.eolang.maven.hash.CommitHashesMap;
 
 /**
- * Object name versioned.
- * This is object name that parses raw sting like:
- * - "org.eolang.text#0.1.0" into "org.eolang.text"
- *   and "4b19944"
- * - "org.eolang.string#a1b2c3d" into "org.eolang.string"
- *   and "be83d9a"
- * Pay attention to that versions transformed into narrow hashes.
- * If a version is not provided - behaves like {@link OnUnversioned}.
+ * Object name with default hash.
+ * If a given object does not contain a version - put the default one.
  *
- * @since 0.30
- * @todo #2376:90min Frontend and backend delimiters differ.
- *  I was confused with the delimiter '#' that we use in {@link OnVersioned} and delimiter which
- *  we use in the frontend. For example:
- *  - "org.eolang.text|0.1.0" - frontend
- *  - "org.eolang.text#0.1.0" - backend
- *  The problem here is that we use  the '|' delimiter on the frontend and '#' in the backend, but
- *  both of them mean the same thing - object name + version.
- *  I believe that we need to use the same symbol in both places, because it will be easier to
- *  understand the code. So, my suggestion to use '|' in both places.
+ * @since 0.29.6
  */
 public final class OnVersioned implements ObjectName {
 
     /**
-     * Delimiter between name and hash in EO object name.
+     * Object with or without hash.
      */
-    public static final String DELIMITER = "#";
+    private final Unchecked<String> object;
```

Comment:
@maxonfjvipon Why do we need to use `Unchecked` here? Can we keep the simple code with `String`?

Reply:
@volodya-lombrozo this change was added [here](https://github.com/objectionary/eo/pull/2386), so it's not a part of the PR. But there's a [reason](https://github.com/objectionary/eo/pull/2391/files/716599c381c2dad36a8b26dc9418bdcd8ddbd434#r1294305053) for using `Unchecked`

Reply:
@maxonfjvipon I see, thanks.